utilities/emr-custom-autoscaler/README.md

```diff
@@ -29,11 +36,74 @@ sudo sh IFautoscaler.sh
 b)To schedule the script as a cron job and execute it every 5 minutes, add the following line to your crontab file:
    */5 * * * * /bin/bash IFautoscaler.sh
 
+### Running on AWS Lambda
+If you prefer to run the autoscaler independent of the cluster, then you can use the `emr-custom-autoscaler.py` python 
+module. This supports all of the same logic as the bash based scaling script, but provides the following additional features:
+* Ability to run in AWS Lambda independent of the Cluster
+* Ability to run from a command line with full configuration support
+* Ability to configure threshold settings above from cli args, environment variables, or via a Dict argument for inclusion in other python modules
+
+When running in this mode, the autoscaler places a timestamping lock file onto Amazon S3 for cooldown logic.
+
+#### Instructions - CLI
+
+Simply clone this repository and then run:
+```bash
+python3 emr-custom-autoscaler.py <options>
+
+options:
+  -h, --help            show this help message and exit
+  --cluster_id CLUSTER_ID 
+                        Cluster ID
+  --config_object CONFIG_OBJECT
+                        Configuration Object (JSON)
+  --config_s3_file CONFIG_S3_FILE
+                        Configuration S3 File
+  --up_threshold UP_THRESHOLD
+                        Threshold for scaling up (YARN Memory Available Percentage)
+  --down_threshold DOWN_THRESHOLD
+                        Threshold for scaling down (YARN Memory Available Percentage)
+  --container_pending_ratio_up_threshold CONTAINER_PENDING_RATIO_UP_THRESHOLD
+                        Threshold for Container Pending Ratio (up)
+  --container_pending_ratio_down_threshold CONTAINER_PENDING_RATIO_DOWN_THRESHOLD
+                        Threshold for Container Pending Ratio (down)
+  --apps_pending_up_threshold APPS_PENDING_UP_THRESHOLD
+                        Threshold for Apps Pending (up)
+  --apps_pending_down_threshold APPS_PENDING_DOWN_THRESHOLD
+                        Threshold for Apps Pending (down)
+  --cooldown_period COOLDOWN_PERIOD
+                        Cooldown period in seconds
+  --ignore_cooldown IGNORE_COOLDOWN
+                        Ignore Cooldown
+  --min_capacity MIN_CAPACITY
+                        Minimum capacity
+  --max_capacity MAX_CAPACITY
+                        Maximum capacity
+  --yarn_memory_scale_factor YARN_MEMORY_SCALE_FACTOR
+                        Scaling factor for YARN Memory Available
+  --container_pending_ratio_scale_factor CONTAINER_PENDING_RATIO_SCALE_FACTOR
+                        Scaling factor for Container Pending Ratio
+  --apps_pending_scale_factor APPS_PENDING_SCALE_FACTOR
+                        Scaling factor for Apps Pending
+  --s3_bucket S3_BUCKET
+                        S3 bucket for log storage
+  --log_level LOG_LEVEL
+                        Log Level
 
+```
 
+Arguments `--cluster_id` and `--s3_bucket` are required.
```

Comment:
Possibly - but how would you propose to deduplicate the possible options if you aren't running locally? Could be many clusters. One thing we could do would be to make it so this can run on N clusters simultaneously.

Reply:
Have a look at the updated logic which now includes arg `for_all_clusters` which will process all RUNNING or WAITING clusters in the region. We probably need to think about tag based restrictions.

Reply:
Thank you for addressing this! I like the tag based approach. This is now resolved.